src/main/java/com/iri/dao3/dao/jdbc/CarDaoImpl.java

```diff
@@ -0,0 +1,237 @@
+                PreparedStatement preparedStatement = connection.prepareStatement(deleteQuery)
+        ) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't mark car in method delete", e);
+        }
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        String insertQuery = "INSERT INTO cars_drivers (driver_id, car_id) "
+                + "VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(insertQuery)
+        ) {
+            preparedStatement.setLong(1, driver.getId());
+            preparedStatement.setLong(2, car.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert in method addDriverToCar", e);
+        }
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        String deleteQuery = "DELETE FROM `cars_drivers` WHERE car_id = ? AND driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(deleteQuery)
+        ) {
+            preparedStatement.setLong(1, car.getId());
+            preparedStatement.setLong(2, driver.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete in method removeDriverFromCar", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectQuery = "SELECT  c.id as id, c.model as model, "
+                + "m.id as manufacturer_id, m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars_drivers cd "
+                + "INNER JOIN cars c ON cd.car_id = c.id "
+                + "INNER JOIN drivers d ON cd.driver_id = d.id "
+                + "INNER JOIN manufacturer m ON c.manufacturer_id = m.id "
+                + "WHERE c.deleted != TRUE AND d.deleted != TRUE AND m.deleted != TRUE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(selectQuery)
+        ) {
+            List<Car> cars = new ArrayList<>();
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(constructCarFromResultSet(resultSet).get());
+            }
+            for (Car car : cars) {
+                car.setDrivers(getAllDriversByCar(car));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't execute getAllByDriver in CarDaoImpl", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Could not get all cars by driver with id " + driverId, e);
```
We should avoid class names in exception messages since they don't provide relevant information

Reply:
ok